app/controllers/api/v1/authentication_controller.rb

```diff
@@ -62,17 +62,13 @@ class AuthenticationController < Api::V1::ApiController
     error 500, "Internal Server Error"
     # Lookup user based on mobile. Validate mobile format first.
     def send_pin
-      mobile = params[:mobile]
-      if (User::HongKongMobileRegExp === mobile)
-        @user = User.find_by_mobile(params[:mobile])
-        @user.send_verification_pin if @user.present?
+      @mobile = params[:mobile]
+      if(@user = valid_user)
+        return render_error(I18n.t('host.invalid'), 403) if !valid_host?
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.